src/cr/cube/crunch_cube.py

```diff
@@ -931,6 +784,37 @@ def as_array(self, include_missing=False, weighted=True, adjusted=False,
         )
         return self._fix_shape(array)
 
+    def _margin(self, axis=None, weighted=True,
+                include_transforms_for_dims=None, prune=False):
+        '''Gets margin if possible, as_array otherwise.
+
+        If a margin is called internally across items dimension, we return
+        as_array(). This can happen when pruning indices, or when calculating
+        zscores. In those cases we need marginals by rows and cols. If any of
+        those happens to be across items dimension, we don't really need to
+        prune (because items will never be pruned). But we still need a result
+        to be able to perform subsequent operations, without explicitly
+        checking if the axes direction is across items dimension each time.
+        '''
+
+        if not self._is_axes_allowed(axis):
+            # In case we encountered axis that would go across items dimension,
+            # we need to return at least some result, to prevent explicitly
+            # checking for this condition, wherever self._margin is used
+            return self.as_array(
```

Comment:
returning counts when requesting margin is a big nono imo

Reply:
I know... That's why the API method will never do that. One other approach is calling this method `_margin_or_counts`. Or something like `base`, or something. I'd much prefer to keep this in a single method to having to write
```
self.margin(axis=0) if self.dim_types[0] != 'categorical_array' else self.as_array()
```

Reply:
but why would you return the counts even? it will produce wrong numbers, there is absolutely no reason in doing that